src/test/groovy/com/bisnode/opa/client/ManagingDocumentSpec.groovy

```diff
@@ -0,0 +1,79 @@
+package com.bisnode.opa.client
+
+import com.bisnode.opa.client.data.OpaDataApi
+import com.bisnode.opa.client.data.OpaDocument
+import com.bisnode.opa.client.rest.ContentType
+import com.github.tomakehurst.wiremock.WireMockServer
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static com.bisnode.opa.client.rest.ContentType.Values.APPLICATION_JSON
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo
+import static com.github.tomakehurst.wiremock.client.WireMock.put
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+
+class ManagingDocumentSpec extends Specification {
+
+    private static int PORT = 8181;
+
+    private OpaDataApi client
+    private String host = "http://localhost:$PORT"
+    private WireMockServer wireMockServer = new WireMockServer(PORT)
+
+    def DOCUMENT = '{"example": {"flag": true}}'
+
+    def setup() {
+        wireMockServer.start()
+        client = OpaClient.builder().opaConfiguration(host).build();
+
+    }
+
+    def cleanup() {
+        wireMockServer.stop()
+    }
+
+    def 'should perform successful policy create or update'() {
+        given:
+          def documentPath = 'somePath'
+          def endpoint = "/v1/data/$documentPath"
+          wireMockServer
+                  .stubFor(put(urlEqualTo(endpoint))
+                          .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON))
+                          .willReturn(aResponse()
+                                  .withStatus(200)
+                                  .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                                  .withBody('{}')))
+        when:
+          client.createOrOverwriteDocument(new OpaDocument(documentPath, DOCUMENT))
+        then:
+          noExceptionThrown()
+          wireMockServer.verify(putRequestedFor(urlEqualTo(endpoint))
+                  .withRequestBody(equalTo(DOCUMENT))
+                  .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON)))
+    }
+
+    @Unroll
+    def 'should handle status code #status correctly'() {
```

Comment:
Test name is not descriptive enough. Maybe `should throw OpaClientException on status code = #status` would be better?